website/content/api-docs/features/consistency.mdx

```diff
@@ -37,15 +64,140 @@ The three read modes are:
+  server agents other than the leader. Other requests will always be handled
+  by the leader no matter the number of Consul servers.
+  The most effective way to increase read scalability is to convert non-`stale`
+  reads to `stale` reads.
+  Only `stale` requests benefit from the enhanced read scalability provided by
+  [read replicas](/docs/enterprise/read-scale) and
+  [redundancy zones](/docs/enterprise/redundancy).
 
-Note that some endpoints support a `cached` parameter which has some of the same
-semantics as `stale` but different trade offs. This behavior is described in
-[agent caching feature documentation](/api/features/caching).
+### Intra-Datacenter Request Behavior
+
+The following diagrams show how the intra-datacenter request path varies
+per consistency mode and the relative trade-offs between level of consistency and performance.
+
+<Tabs>
+<Tab heading="Stale reads">
+
+[![stale consistency mode read request path](/img/consistency-modes/consistency-mode-stale.png)](/img/consistency-modes/consistency-mode-stale.png)
+
+</Tab>
+
+<Tab heading="Default reads">
+
+[![default consistency mode read request path](/img/consistency-modes/consistency-mode-default.png)](/img/consistency-modes/consistency-mode-default.png)
+
+</Tab>
+<Tab heading="Consistent reads, all writes">
+
+[![consistent consistency mode read, any write request path](/img/consistency-modes/consistency-mode-consistent.png)](/img/consistency-modes/consistency-mode-consistent.png)
+
+</Tab>
+</Tabs>
+
+### Cross-Datacenter Request Behavior
+
+When making a request across federated Consul datacenters, requests are forwarded from
+a local server to any remote server. Once in the remote datecenter, the request path
+is the same as a [local request with the same consistency mode](#intra-datacenter-request-behavior).
+
+[![consistency mode behavior across Consul datacenters](/img/consistency-modes/consistency-mode-cross-datacenter.png)](/img/consistency-modes/consistency-mode-cross-datacenter.png)
+
+## Usage
+
+### Consul DNS Queries: Defaults and Overriding
+
+When DNS queries are issued to [Consul's DNS interface](/docs/discovery/dns),
+Consul uses the `stale` consistency mode by default when interfacing with its
+underlying Consul service discovery HTTP APIs
+([catalog](/api-docs/catalog), [health](/api-docs/health), and [query](/api-docs/query)).
+Though this default can be changed, doing so is not recommended.
+
+The consistency mode underlying Consul DNS queries cannot be overridden on a
+per-request basis.
+
+#### Changing the Default Consistency Mode
+
+If stronger consistency is required at the expense of increased latency,
+set [`dns_config.allow_stale`] to `false` to make Consul use the
+`default` consistency mode instead of `stale`
+when interfacing with the underlying Consul service discovery HTTP APIs.
+
+#### Limiting Staleness (Advanced Usage)
+
+~> **Not recommended**: we do not recommend using [`dns_config.max_stale`]
```

Comment:
@banks : can you check the accuracy of this subsection?

Reply:
I agree that the current framing is "here's a feature that sounds great but we tell you not to use". I think that is actually what we want to communicate clearly (that people probably shouldn't use it). I could see someone then asking the reasonable question "well, why was this ever added in the first place?", but I don't think the history matters to much to them.

Maybe the simplest solution is to change the bolded part to:
```suggestion
~> **Use with caution**: we do not recommend using [`dns_config.max_stale`]
```

I considered this initially, but thought it might not be strong enough to discourage usage at-a-glance. What do you think?

Reply:
I'm not sure how to word that @trujillo-adam. It's not so much that setting that causes overload, it's more that in general the most likely times it will have any effect are times where it will tend to do more harm than good. It's one of those things that sounds like a good idea and you can contrive situations where it is desirable behaviour to be able to limit staleness of responses, but in practice it's much more likely to cause problems than actually have the benefit you might assume.

I tried to capture that nuance but I can see it's perhaps a bit wordy/narrative focussed for reference docs.

@jkirschner-hashicorp do you think a reasonable compromise between not spelling out the history and explaining the recommendation would be to use words like: `Warning: setting this has caused undesirable side effects such as causing high server load to turn into a prolonged outage in some cases. We currently don't recommend setting it.`?

Reply:
Ah, OK. I think a `Caution` label is an appropriate label for the callout because while this is something you can do, you should be wary of doing it. `Warning` might sound like failure is imminent. Maybe we shouldn't allow people to do this?

Jared, sure, something like what you have is probably clear. Here's my attempt:

```
`Caution`: Configuring the `dns_config.max_stale` to limit staleness may result 
in a prolonged outage if your server becomes overloaded. We do not recommend 
configuring this setting if the potential for overloading your servers exists.
```
I like "prolonged outage" because it's something concrete, but I guess we can say something more general like the original language you provided " . . . may exacerbate issues associated with high server loads . . ."

Reply:
> Maybe we shouldn't allow people to do this?

Honestly I'd prefer to just remove the feature but I think that's a longer discussion as some will be using it and not have had issues etc.

I like your suggested wording @trujillo-adam, although I think the second sentence more or less sums up the problem here - it's practically impossible for an operator to guarantee that their servers won't get overloaded so this warning actually amounts to "never use this" 😄! I prefer your first sentence though. How about if we shift the focus of the second sentence to when you _maybe_ should:

> ```
> `Caution`: Configuring the `dns_config.max_stale` to limit staleness may result in a prolonged outage if your server becomes overloaded. We do not recommend configuring this setting unless your application's correctness would be compromised if results are too stale and you'd rather be unavailable than incorrect.

Or something, I'm sure you can make that last sentence more eloquent but I'm trying to get at the real tradeoff here: is it better to be down or a bit more stale than you would like.

Reply:
Here's what I'm thinking

---

**Caution**: Configuring [`dns_config.max_stale`] may result in a prolonged outage if your Consul servers become overloaded. We do not recommend using this setting to limit staleness unless it would be preferable for your services to be unavailable than to receive results that are too stale. If bounded result consistency is required by a service, consider modifying the service to use `consistent` service discovery HTTP API queries instead of DNS lookups.